lib/cfg.c

```diff
@@ -620,6 +620,9 @@ cfg_free(GlobalConfig *self)
   if (self->original_config)
     g_string_free(self->original_config, TRUE);
 
+  if (self->file_list)
+    g_list_free(self->file_list);
```

Comment:
The individual paths should be freed too (they are g_strdupped now). `g_list_free_full(self->file_list, g_free)` could be used.

You do not need to check if `self->file_list` is null or not. List functions in glib handle it correctly.

Reply:
Ok, I will make that change.